rclc_parameter/src/rclc_parameter/parameter_server.c

```diff
@@ -827,6 +1545,134 @@ rclc_parameter_server_init_service(
   return rclc_service_init(service, node, srv_type, get_service_name, &rmw_qos_profile_parameters);
 }
 
+rcl_ret_t rclc_add_parameter_description(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name, const char * parameter_description,
+  const char * additional_constraints)
+{
+  if (parameter_server->low_mem_mode) {
+    return RCLC_PARAMETER_UNSUPORTED_ON_LOW_MEM;
+  }
+
+  size_t index = rclc_parameter_search_index(&parameter_server->parameter_list, parameter_name);
+
+  if (index >= parameter_server->parameter_list.size) {
+    return RCL_RET_ERROR;
+  }
+
+  ParameterDescriptor * parameter_descriptor = &parameter_server->parameter_descriptors.data[index];
+
+  // Set parameter description
+  if (!rclc_parameter_set_string(&parameter_descriptor->description, parameter_description)) {
+    return RCL_RET_ERROR;
+  }
+
+  // Set constrain description
+  if (!rclc_parameter_set_string(
+      &parameter_descriptor->additional_constraints,
+      additional_constraints))
+  {
+    return RCL_RET_ERROR;
+  }
+
+  return RCL_RET_OK;
+}
+
+rcl_ret_t rclc_set_parameter_read_only(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name, bool read_only)
+{
+  if (parameter_server->on_callback) {
+    return RCLC_PARAMETER_DISABLED_ON_CALLBACK;
+  }
+
+  size_t index = rclc_parameter_search_index(&parameter_server->parameter_list, parameter_name);
+
+  if (index >= parameter_server->parameter_list.size) {
+    return RCL_RET_ERROR;
+  }
+
+  ParameterDescriptor * parameter_descriptor = &parameter_server->parameter_descriptors.data[index];
+
+  // Set flag
+  parameter_descriptor->read_only = read_only;
+
+  return RCL_RET_OK;
+}
+
+rcl_ret_t rclc_add_parameter_constraint_double(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name, double from_value,
+  double to_value, double step)
+{
+  if (parameter_server->on_callback) {
```

Comment:
Please add NULL pointer checks for all pointer arguments.

Reply:
Added a8431b4ed7f6ee02ab27705d86f60b30eae6b052